scripts/DCOSWindowsAgentSetup.ps1

```diff
@@ -184,13 +184,33 @@ function Install-MetricsAgent {
     }
 }
 
-function Update-Docker {
-    $dockerHome = Join-Path $env:ProgramFiles "Docker"
+function Install-Docker {
+    $service = Get-Service "Docker" -ErrorAction SilentlyContinue
+    if($service) {
+        Stop-Service "Docker"
+        sc.exe delete "Docker"
+        if($LASTEXITCODE) {
+            Throw "ERROR: Failed to delete existing Docker service"
+        }
+    }
+    $dockerRegKey = "HKLM:\SYSTEM\CurrentControlSet\Services\EventLog\Application\docker"
+    if(Test-Path $dockerRegKey) {
+        Remove-Item -Force $dockerRegKey
+    }
+    $installDir = Join-Path $env:ProgramFiles "Docker"
+    if(!(Test-Path $installDir)) {
+        New-Item -ItemType "Directory" -Path $installDir
+    }
```

Comment:
Could those manually uninstalling process be replaced by the following powershell commands?

Uninstall-Module dockerprovider
Uninstall-package docker

Reply:
Hmm, this assumes that Docker is always installed via `Install-Package` / `Install-Module`, right ?

Sometimes the user can simply install Docker manually by executing the PowerShell code from [here](https://docs.docker.com/install/windows/docker-ee/#docker-universal-control-plane-and-windows), section *Use a script to install Docker EE*.

Nonetheless, this piece of code simply installs Docker if it's not installed. And if it's already installed, it replaces the old files.

I think this should be fine in both cases: either you installed Docker manually or via `Install-Package`.

Reply:
I was wondering how a Windows node without Docker installed got into a DCOS cluster.
The Windows server with container enabled will have Enterprise Docker installed by default (such as WindowsServer20160Container, RS3 images...) -- via about powershell setup scripts.

If it's from a custom VHD, it should be prepared with container-enabled , and Docker installed before putting it into any DCOS cluster setup.

Reply:
Bruce had a custom image without `Docker` in it and he suggested that it would be nice to have all the requirements installed by the script.

However, you need to install the `Containers` feature as well and that requires a reboot and it would be a problem for the rest of the script. So, we agreed that the `Containers` feature and `Docker` should be installed on every custom image and documentation should reflect that.

This was a leftover PR with:
- Install Docker if it's not there
- Fix for `Get-MasterIPs` function

I'm favor of removing the commit https://github.com/dcos/dcos-windows/pull/31/commits/1d65f7ca6d1b3c0506fad57fb302be71a282daf0 and keep only the other one with the fix. What do you think ?

Reply:
I totally agree with you on this. 

"A custom VHD needs be prepared with Hyper-V and Container-enabled , and Docker installed"

Let's remove commit 1d65f7c

The rest of the changes look good.

Reply:
Done. The PR now includes only the one-line fix commit.